### PR TITLE
Plots (added two functions)

### PR DESCRIPTION
**Pull request by danielli97**
Added plot_loadings and plot_pc_variance (both tested against mtcars)
plot_loadings takes a sample and shows (most to least, or specified number) of its contributions to principal components
plot_pc_variance shows total percentage contribution of principal components (can show which PC to start at and how many)

### CODE REVIEW

File: R/plots.R [hunk 1]
```diff
@@ -125,6 +125,133 @@ plot_pca <- function(obj,
   p
 }
 
+
+#' Plot Loadings and Interpretations
```
Comment (pimentel) [inline R/plots.R]: does this build without a proper description? there should be another line or more below with a more thorough description. make sure you are calling `document()` every time you introduce new documentation

File: R/plots.R [hunk 2]
```diff
@@ -125,6 +125,133 @@ plot_pca <- function(obj,
...
+#' Plot Loadings and Interpretations 
+#' @param obj a \code{sleuth} object
+#' @param use_filtered if TRUE, use filtered data. otherwise, use all data
+#' @param PC is the (string) name of which principal component to view genes contribution to that PC
+#' @param gene_count user input on how many genes to display, otherwise default is 5
+#' @return a ggplot object
+#' @export
+plot_loadings <- function(obj, 
+  use_filtered = TRUE,
+  PC = NULL, #this is a string of the PC actual name
```
Comment (pimentel) [inline R/plots.R]: don't use caps for variable names that aren't global

File: R/plots.R [hunk 3]
```diff
@@ -125,6 +125,133 @@ plot_pca <- function(obj,
...
+  gene_count = NULL,
+  bool = FALSE,
```
Comment (pimentel) [inline R/plots.R]: this needs a more descriptive name, and also needs to be documented above

File: R/plots.R [hunk 4]
```diff
@@ -125,6 +125,133 @@ plot_pca <- function(obj,
...
+  #sort loadings vector to obtain highest contribution
+  loadings <- sort(loadings, decreasing = TRUE)
+  names <- names(loadings)
+
+  dat <- data.frame(pc = names, loadings = loadings)
+  dat$pc <- factor(dat$pc, levels = unique(dat$pc))
+
+  p <- ggplot(dat, aes(x = pc, y = loadings)) 
+  p <- p + geom_bar(stat = "identity")
+  p <- p + xlab("Principal Components") + ylab("Contribution")
```
Comment (pimentel) [inline R/plots.R]: only 1 layer per line please

File: R/plots.R [hunk 5]
```diff
@@ -125,6 +125,133 @@ plot_pca <- function(obj,
...
+  } else {
+    p <- p + ggtitle(title)
+  }
+
+  p
+
+}
+
+#' Plot PC variances retained by percentage with option to compare specified PC
+#'
```
Comment (pimentel) [inline R/plots.R]: again, documentation here. see the other plots for examples

File: R/plots.R [hunk 6]
```diff
@@ -125,6 +125,133 @@ plot_pca <- function(obj,
...
+#' @param pca_number user input on how many PC to display, otherwise default is 5
+#' @param bool determines scaling
+#' @param (integer) PC_relative gives the option to compare subsequent principal
+#' components and their contributions
+#' @return a ggplot object
+#' @export
+plot_pc_variance <- function(obj, 
+  use_filtered = TRUE,
+  pca_number = NULL,
+  bool = FALSE,
```
Comment (pimentel) [inline R/plots.R]: again, 8 a more descriptive variable name end needs to be documented better

File: R/plots.R [hunk 7]
```diff
@@ -125,6 +125,133 @@ plot_pca <- function(obj,
...
+#' @param use_filtered if TRUE, use filtered data. otherwise, use all data
+#' @param pca_number user input on how many PC to display, otherwise default is 5
+#' @param bool determines scaling
+#' @param (integer) PC_relative gives the option to compare subsequent principal
+#' components and their contributions
+#' @return a ggplot object
+#' @export
+plot_pc_variance <- function(obj, 
+  use_filtered = TRUE,
+  pca_number = NULL,
```
Comment (pimentel) [inline R/plots.R]: keep the variable names consistent

File: R/plots.R [hunk 8]
```diff
@@ -125,6 +125,133 @@ plot_pca <- function(obj,
...
+    mat <- spread_abundance_by(obj$obs_norm_filt, units)
+  } else {
+    mat <- spread_abundance_by(obj$obs_norm, units)
+  }
+
+  pca_calc <- prcomp(mat, scale = bool) #PCA calculations 
+
+  #computation
+  eigenvalues <- (pca_calc$sdev)^2  
+  var <- eigenvalues*100/sum(eigenvalues)
```
Comment (pimentel) [inline R/plots.R]: `var` is already the name of a built in R function. do not overwrite it. also, use a more descriptive name

File: R/plots.R [hunk 9]
```diff
@@ -125,6 +125,133 @@ plot_pca <- function(obj,
...
+  var2 <- eigenvalues*100/sum(eigenvalues) #because i suck at coding
```
Comment (pimentel) [inline R/plots.R]: address comments from other thread here

File: R/plots.R [hunk 10]
```diff
@@ -125,6 +125,133 @@ plot_pca <- function(obj,
...
+  
+  #from here to ....
+  if (!is.null(pca_number)) {
+    colsize <- pca_number
```
Comment (pimentel) [inline R/plots.R]: keep consistent with variable name conventions (using snake_case)

File: R/plots.R [hunk 11]
```diff
@@ -125,6 +125,133 @@ plot_pca <- function(obj,
...
+    var <- var[1:pca_number]
+  } else {
+    colsize <- 5 #default 5
+    var <- var[1:5] #default 5
+  }
+  pc_asdf <- data.frame(PC_count = 1:colsize, var = var) #order here matters
```
Comment (pimentel) [inline R/plots.R]: capitalization here, what is `_asdf`?

File: R/plots.R [hunk 12]
```diff
@@ -125,6 +125,133 @@ plot_pca <- function(obj,
...
+  #...here is for comparison of variance for subsequent PC's
+
+  #here is comparison per given principal component
+  if(!is.null(PC_relative)) {
+    pc_asdf <- data.frame(PC_count = 1:length(eigenvalues), var = var2) #because i suck at coding (var2 stupid shit)
```
Comment (pimentel) [inline R/plots.R]: this comment should not be here

**Comment by pimentel** [comment]
Overall, looks pretty good. I commented mostly on style. After you fix that, I will likely make another pass and actually run the code (and comment on the shiny stuff). few notes:

- do not put exploratory code or figures in the repository
- please keep comments relevant (telling the reader that you suck at coding isn't particularly informative)
- please work on improving your commit messages:

http://chris.beams.io/posts/git-commit/
https://github.com/erlang/otp/wiki/Writing-good-commit-messages

- please go back and change the messages that basically don't say anything (`asdf`). I am fine with you saying 'work in progress (wip)', but again please try to keep the commit messages meaningful

**Comment by Blahah** [comment]
Just a thought - might be good to squash the commits to one per feature. That will make the history more readable (and avoids the need for editing all the commit messages).

**Pull request closed without merging**